Update ruff to version 0.5.2 and workflows update

Smartappli opened the pull request:
Workflow updated

qubvel commented:
Hi @Smartappli , thanks for the opening PR!
Let me know when its ready 🙂

Smartappli commented:
> Hi @Smartappli , thanks for the opening PR! Let me know when its ready 🙂

Hi @qubvel, its ready

qubvel commented:
Can you please alse update ruff in setup.py?

qubvel commented:
One more note here, can you please pin timm version back? It's going to be updated in another PR, thanks.

Smartappli commented:
@qubvel ready to merge

qubvel commented on ruff.toml:
```diff
@@ -0,0 +1,61 @@
+]
+
+# Same as Black.
+line-length = 88
```
I would prefer a longer line, e.g. 120 chars

Smartappli replied:
done

qubvel commented on ruff.toml:
```diff
@@ -0,0 +1,61 @@
+# Exclude a variety of commonly ignored directories.
```
Can you please tell what was the motivation for adding this config?

qubvel commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
-torchvision>=0.5.0
+torchvision>=0.10.0
```
Any reason to do that?

qubvel reviewed (commented):
Thanks for working on this, a few more comments below. Moreover, CI fails due to changed ruff configuration

qubvel commented:
Sorry, probably there was a kind of misunderstanding, I thins exclude is necessary, but I asked about what was the motivation to add `ruff.toml`?

qubvel commented:
Sorry, I cannot merge it this way because it breaks CI. I suggest moving ruff config change to a separate PR and keeping here just ruff version change to be able to pass the CI checks.

In **next** PR the config might be created and changed, then ruff formatting applied to reflect config changes.

Smartappli commented:
> Sorry, I cannot merge it this way because it breaks CI. I suggest moving ruff config change to a separate PR and keeping here just ruff version change to be able to pass the CI checks.
> 
> In **next** PR the config might be created and changed, then ruff formatting applied to reflect config changes.

reverted

qubvel reviewed (approved):
Looks good, thanks for iterating!
